pkg/kubelet/kubelet.go

```diff
@@ -1524,6 +1524,8 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 		if !pcm.Exists(pod) && !firstSync {
 			if err := kl.killPod(pod, nil, podStatus, nil); err == nil {
 				podKilled = true
+			} else {
+				klog.Errorf("killPod for pod %q failed: %v", format.Pod(pod), err)
```

Comment:
I wonder if in this case it should return as well. Alternative is that pod is terminating, runtime failed to kill the pod, and  will not be restarted.

Also, interesting thing here is that condition was added https://github.com/kubernetes/kubernetes/pull/39606 seemingly for the case when `podStatus` is `nil`. So it will not even attempt to call into the runtime